### PR TITLE
Finish all API auth and test

### PR DESCRIPTION
**Pull request by qmo1222**

### CODE REVIEW

File: src/clue.js [hunk 1]
```diff
@@ -1,162 +1,277 @@
...
 	var ret = new Object;
-	ret.uid = parseInt(req.body.operator_uid);
+	ret.uid = operator_uid;
 	ret.object = 'clue';
 	ret.action = 'create';
 
-	var check = !isNaN(ret.uid) && (clue.content != undefined);
-	if (check) {
-		connection.query('INSERT INTO clue SET ?', clue, function(err, result){
+	fire.on('check', function(){
```
Comment (jwwandy) [inline src/clue.js]: Seems like check is used everytime

Comment (jwwandy) [inline src/clue.js]: Why we have so many spaces for identation

Comment (jwwandy) [inline src/clue.js]: Check event might be useless

Comment (jwwandy) [reply src/clue.js]: Why we have so many spaces for identation

**Review by jwwandy — CHANGES_REQUESTED** [review]

**Comment by qmo1222** [comment]
Pull request so far including :
* Add test for tool image upload
* Add auth level and token
* Modify all tests for auth level and token
* Filter future mission with time for frontend reading
* Fix database timezone problem
* Put db querying into *object.js*

**Comment by qmo1222** [comment]
Function added :
* Valid area check

**Review by jwwandy — APPROVED** [review]
Great Job!

**Pull request merged**